include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -103,6 +103,15 @@ class MGConstrainedDoFs : public Subscriptor
     const std::set<types::boundary_id> &boundary_ids,
     const ComponentMask &               component_mask = ComponentMask());
 
+  /**
+   * Set user defined contraints to be used only in the MGTransferMatrixFree
+   * class. The user can add multiple constraints, but they will be
+   * merged such that the current merge will win any conflicts.
```

Comment:
suggestion:
```
   * Set user defined constraints to be used on level @p level.
   *
   * The user can add ...
   * @note This is currently only implemented for MGTransferMatrixFree.
```

What do you mean by "can add"? Do you mean "The function can be called more than once per level and constraints will be merged with the existing constraints"? If yes, how do you reset the constraints? Why is the function not called add_user_constraints?

Please also document how this interacts with Dirichlet constraints.